app/scenes/CollectionPermissions/AddGroupsToCollection.tsx

```diff
@@ -69,84 +75,81 @@ class AddGroupsToCollection extends React.Component<Props> {
+            {t("Create a group")}
+          </ButtonLink>
+          .
+        </Text>
+      ) : null}
 
-        <Input
-          type="search"
-          placeholder={`${t("Search by group name")}…`}
-          value={this.query}
-          onChange={this.handleFilter}
-          label={t("Search groups")}
-          labelHidden
-          flex
-        />
-        <PaginatedList
-          empty={
-            this.query ? (
-              <Empty>{t("No groups matching your search")}</Empty>
-            ) : (
-              <Empty>{t("No groups left to add")}</Empty>
-            )
-          }
-          items={groups.notInCollection(collection.id, this.query)}
-          fetch={this.query ? undefined : groups.fetchPage}
-          renderItem={(item: Group) => (
-            <GroupListItem
-              key={item.id}
-              group={item}
-              showFacepile
-              renderActions={() => (
-                <ButtonWrap>
-                  <Button onClick={() => this.handleAddGroup(item)} neutral>
-                    {t("Add")}
-                  </Button>
-                </ButtonWrap>
-              )}
-            />
-          )}
-        />
-        {can.createGroup && (
-          <Modal
-            title={t("Create a group")}
-            onRequestClose={this.handleNewGroupModalClose}
-            isOpen={this.newGroupModalOpen}
-          >
-            <GroupNew onSubmit={this.handleNewGroupModalClose} />
-          </Modal>
+      <Input
+        type="search"
+        placeholder={`${t("Search by group name")}…`}
+        value={query}
+        onChange={handleFilter}
+        label={t("Search groups")}
+        labelHidden
+        flex
+      />
+      <PaginatedList
+        empty={
+          query ? (
+            <Empty>{t("No groups matching your search")}</Empty>
+          ) : (
+            <Empty>{t("No groups left to add")}</Empty>
+          )
+        }
+        items={groups.notInCollection(collection.id, query)}
+        fetch={query ? undefined : fetchGroups}
+        renderItem={(item: Group) => (
+          <GroupListItem
+            key={item.id}
+            group={item}
+            showFacepile
+            renderActions={() => (
+              <ButtonWrap>
+                <Button onClick={() => handleAddGroup(item)} neutral>
+                  {t("Add")}
+                </Button>
+              </ButtonWrap>
+            )}
+          />
         )}
-      </Flex>
-    );
-  }
+      />
+      {can.createGroup ? (
+        <Modal
+          title={t("Create a group")}
+          onRequestClose={handleNewGroupModalClose}
+          isOpen={newGroupModalOpen}
+        >
+          <GroupNew onSubmit={handleNewGroupModalClose} />
+        </Modal>
+      ) : null}
+    </Flex>
+  );
 }
 
 const ButtonWrap = styled.div`
   margin-left: 6px;
 `;
 
-export default withTranslation()(withStores(AddGroupsToCollection));
+export default AddGroupsToCollection;
```

Comment:
```suggestion
export default observer(AddGroupsToCollection);
```

Needs wrapping in observer as the component is reading MobX state – otherwise it won't react if underlying data changes.

Reply:
Interested in knowing how do you know by looking at props if something is reading from mobx state or not? 
I haven't used mobx so I don't know the difference.